### ISSUE TITLE
Transmission project is abandoned?

### ISSUE THREAD

**Issue by Coeur** [transmission/transmission#7058]
### What is the issue?

I actually just noticed that the admins of Transmission are more active than I am on GitHub... but they contribute on other projects (Electron for one, private projects for the other). I previously incorrectly assumed they were absent from internet for personal reasons (https://github.com/transmission/transmission/issues/6813#issuecomment-2123596197).

So in reality, they are active GitHub users, yet they never addressed the requests for extra maintainers to the project, deny receiving requests for write access (https://github.com/transmission/transmission/issues/6813#issuecomment-2138456207), while barely reviewing/merging anymore on Transmission.

Oh, I'm giving up waiting. I may move to [libtorrent](https://github.com/arvidn/libtorrent), which has seemingly only a single active admin (arvidn), but that's one more than here.

### Which application of Transmission?

None

**Comment by Pentaphon** [comment]
If what you say is true, then yes, we need another leader to keep the project going. The PR's are in the 100s. 

>Oh, I'm giving up waiting. I may move to [libtorrent](https://github.com/arvidn/libtorrent), which has seemingly only a single active admin (arvidn), but that's one more than here.

I don't think the community will benefit from having just one active torrent lib out there. I think libtransmission and libtorrent-rasterbar should both be active. I'm in favor of escalating efforts to get the Transmission admins to listen to reason and allow somebody to move the project along themselves when they are absent or uninterested.

**Comment by nevack** [comment]
I share your concerns about the future of the project, @Coeur 

<details>
<summary>/offtopic</summary>

I no longer have the motivation to make any improvements or fixes, knowing that the pull request will just get stuck in limbo indefinitely.

Honestly, I can't imagine where @tearfur finds the strength to keep digging into and fixing problems 😞

</details>

The project has gathered a great community of technically skilled people, each in their own niche. They are willing to contribute in any way they can. But everything is hindered by the sheer impossibility of making necessary and often critical changes.

As I see the situation and understand the project's release cycle, there is an opportunity to fork it and continue active development there. However, it would be impossible to convince distribution maintainers and users to trust the fork as the main source of truth, like they did with the original project.

At the very least, we could attempt to create such a fork while there's still enthusiasm and energy.
Maybe someone would be interested in using new version where performance might be better, bugs get fixed more quickly, and new features are added.
This solution could prevent us from getting stuck in this uncertain situation.

<details>
<summary>/offtopic</summary>

I'm willing to buy Apple Developer license and maintain macOS distribution, at least.

</details>

I really hope @ckerr will step up sooner or later with some kind of statement about current situation in leadership+maintainership.

**Comment by Pentaphon** [comment]
> At the very least, we could attempt to create such a fork while there's still enthusiasm and energy.

I would highly advise against this. This is something that should always be on the table, but only as a last resort. We'd risk fracturing the project and creating trust issues that may never be resolved, and we'd end up at the same place or worse.

I don't know why Charles and Mike aren't communicating with us, but surely they know they have to start transferring power before this recent surge of activity dies down or we risk losing that and have the project languish for another decade. I am hopeful that they realize that there is such a thing as being too protective of the project and do what is best for it. They can't wait until there is only 1 of them standing to start looking for another admin. There just needs to be somebody who is always moving things along and I hope they understand that this is all people are asking for.

**Comment by tearfur** [comment]
> Honestly, I can't imagine where @tearfur finds the strength to keep digging into and fixing problems 😞

@nevack After slogging through the bigger half of this year without much response at all from the maintainers, I now just treat this project as a playground while I can, without too much of a care about making progress (it's not like I have the rights to do so anyway 🤷🏻‍♂️).

I started contributing to libtransmission because I saw code that's clearly been slowly built up with finely thought out steps. The previous authors strived their best to write out each routine only once, organised them into categories (aka classes in C++, or prefixed functions back in the C days) such that they can be reused to their max. Honestly it's really pretty code in my eyes, so I wanted to ride along the transition phase from C to C++.

Having said that, I'm not optimistic about the future of this project either. There's still so much to do but the circumstances are not great. The `4.0.6` tarball submodules staying unfixed nearly 3 months after release is a big red flag for me.

> The project has gathered a great community of technically skilled people, each in their own niche. They are willing to contribute in any way they can. But everything is hindered by the sheer impossibility of making necessary and often critical changes.

Totally agree, it's a shame that the project is in such a slump despite that.

**Comment by mikedld** [comment]
> Electron for one, private projects for the other

Both connected to their day jobs.

I can only speak for myself, so… I'm trying to find time sporadically but I'm really not comfortable reviewing anything but GTK and Qt clients, while most of active contributors' PRs go the other way. I look at libtr PRs from time to time but that's not exactly productive. I also have somewhat pedantic attitude towards the changes (as some of you may have noticed) making it hard for me to accept some of the PRs — doesn't help with encouraging contributions.

**Comment by arcticfoxwp** [comment]
> > Electron for one, private projects for the other
> 
> Both connected to their day jobs.
> 
> I can only speak for myself, so… I'm trying to find time sporadically but I'm really not comfortable reviewing anything but GTK and Qt clients, while most of active contributors' PRs go the other way. I look at libtr PRs from time to time but that's not exactly productive. I also have somewhat pedantic attitude towards the changes (as some of you may have noticed) making it hard for me to accept some of the PRs — doesn't help with encouraging contributions.

Mike, why don't you cut the BS and stop gatekeeping this project by granting write to other pers who are putting in their time and effort to contribute.

This initiative will help further the project, and has been supported by everyone. I can't find a single person who opposes it, except the few who hold they keys to the castle.

**Comment by mikedld** [comment]
@arcticfoxwpg blocked for 3 days to cool off. Let's be constructuve please.

**Comment by Pentaphon** [comment]
>Both connected to their day jobs

No offense, @mikedld but it doesn't matter if any admin has a day job. The whole point of what people have issue with here is that the admins should make it a priority to find somebody who can keep the project going when they can't do so for long periods of time. 

I understand that the admins may be protective of this project but there's reasonable people here who will follow your instructions and only approve bugfixes and optimizations rather than any major UI or functionality changes so the project doesn't turn into something you don't want it to become. I think as long as the client is getting bugfixes and optimizations in new releases on a regular basis, the community will be happy. 

I implore all admins reading this to please make a public commitment to finding somebody who can keep this project going smoothy as soon as possible. You simply cannot wait until there is only 1 admin standing to find another person. This project needs somebody who can keep up with all the PRs before all those enthusiastic contributors decide this project is no longer worth their time and energy.

**Comment by Coeur** [comment]
> but I'm really not comfortable reviewing anything but GTK and Qt clients

Recommendation.
- Mike keeps GTK and Qt write ownerships
- **Tearfur** is added to help Charles for Core write ownerships
- either myself (**Coeur**) or **Nevack** is added for macOS write ownerships
- **Rukario** could be added for Web write ownerships

![Pod-24](https://github.com/user-attachments/assets/c17d42eb-e727-400e-a225-0b228edaa051)

**Comment by bcomnes** [comment]
Just hopping in here to point out aggressive accounts (who can't even be bothered to set up an avatar) pushing for new contributors is how XZ was back-doored. 

Transmission 4.x has a ton of great work in it. Highly appreciative of the project and respect the pace the leadership chooses to take.

**Comment by bbsixzz** [comment]
Just do it for a point release, if it doesn't please you revoke access and stay the slow course and point to this as a tried path that didn't work out.

Do nothing and you lose by default having a displeased user base and contributors, the risk of a backdoor is a slippery slope fallacy. 

The real risk is right in front of us, right now.

**Comment by bcomnes** [comment]
A point release would be nice to due to the recent tracker bannings of 4.0.6, but slow pace is the norm in this project and it has been completely fine for the health of the project. In fact things have been moving faster lately than they have over the last 10 years. See:

<img width="907" alt="Screenshot 2024-08-13 at 3 48 26 PM" src="https://github.com/user-attachments/assets/4f7ced24-c765-430c-a33e-e13d7882d128">

Not saying this is what is happening here, but a pushy contributor backed up by a gaggle of aggressive, anonymous accounts pressuring an unpaid maintainer of a project to elevate contributor access for the sake of project pace is a proven social engineering tactic that lead to ssh getting backdoor in Debian via xz. It's not hypothetical. It's not a fallacy. Anyone coming into a thread to put aggressive pressure on leadership to change out access controls should take a step back and understand the conditions and hazard they are contributing to. Open source doesn't imply open access or open governance. 

As for the maintainers, please vet anyone you give elevated access to. Thanks for all of the hard work over the years, it is appreciated.

**Comment by Pentaphon** [comment]
> Just hopping in here to point out aggressive accounts (who can't even be bothered to set up an avatar) pushing for elevated commit access to newer contributors is how XZ was back-doored.

1. Anybody can set up a duck avatar if they wanted. 
2. Before your comment, everybody has been a regular contributor or commenter. 
3. Nobody is telling the admins to blindly give anybody elevated commit access. Everybody should be vetted. 
4. Transmission is not xz
5.  I've personally always told the admins to be cautious and take every measure they can to ensure the project's security.
6. This thread has a valid complaint that has been brewing for years now. This is not something that appeared overnight.

**Comment by bcomnes** [comment]
The issue isn't purely lack of avatar. It's low identity accounts. Who are you. Where are located? What is your full name? Do have a history of meeting with other real people at events who could vet for you right this moment? 

Compounding the issue is uniform entitlement on display here. Transmission is an 18+ year project and you just show up a couple years ago, and now demand merge rights because the person with followthrough isn't available on your schedule? 🚩🚩🚩🚩

**Comment by Pentaphon** [comment]
> The issue isn't purely lack of avatar. It's low identity accounts. Who are you. Where are located? What is your full name? Do have a history of meeting with other real people at events who could vet for you right this moment?

1. The same could be said for you. Anybody can provide a name on Github. It is up to the admins to vet people privately.
2. I am not the person asking for write access. 
3. I am merely expressing my opinion and the admins can do with that as they wish. Same as you.
4. Nobody is demanding merge rights. If nobody new gets them, so be it. However, I do think the community should be informed as to whether Tearfur got a yes or not for his request for write access he made months ago. 
5. I think people just want better communication and for the project to be managed more efficiently so they don't feel their PRs are in limbo. The contributors keep describing the project as difficult to communicate and exchange with. 
6. I think the first step to take is to open up a private channel for contributors and admins to chat privately, but even that has yet to be accomplished for over a year now.

**Comment by bbsixzz** [comment]
@becomes I hear what you're saying but just stop with the fearmongering as an excuse for everything, including poor communication.

Just do something, say yay or nay and explain why—don't just sit on it and let it brew when there is clear interest and worry.

If a decision can't be made now tell us when.

**Comment by Tharn** [comment]
It would be madness not to give write access to tearfur who, as far as I can tell, has been the single reason that the new Transmission is moving in the right direction with performance optimizations. This is not a case of XZ. This is clearly seeing people who are active on a whole different level to what the core maintainers are able to do, and then letting them starve on a long leash.

Yes, Transmission is an 18 year project which throughout its life has moved at a pretty slow pace. That was *fine* until it rebased with 4.0. You can't just leave it slogging along with these glaring issues after such a pivotal change. I still have not updated my main instance from 3.0 because of the performance regressions.

**Comment by fetzu** [comment]
@Tharn is making a great point: Coeur and tearfur have done tremendous work on the project, and this is clearly not a XZ situation (and that's without taking into account that Transmission is not nearly an efficient attack vector as XZ). To decide wether they should be given write and/or releasing permissions is up to the maintainers, but their contributions and loyalty to the project should definitely not be questioned.

**Comment by Pentaphon** [comment]
> It would be madness not to give write access to tearfur who, as far as I can tell, has been the single reason that the new Transmission is moving in the right direction with performance optimizations.

Let's pull back a bit from this. Tearfur has been an awesome contributor, yes, but he should be vetted like everybody else. It would be madness to suggest that anybody who does great work on a project just automatically get write access. It is up to the admins to decide whether he is trustworthy, carefully and privately. All we can ask is that the admins seriously consider his request and vet him to the best of their ability.

**Comment by Tharn** [comment]
That should always be the case and I'm not suggesting otherwise. It's part of the process. As long as the process is initiated, you know?

**Comment by bcomnes** [comment]
It's incredibly juvenile to publicly call into question the health of a project that you supposedly care about (this does damage to the project), say its in a "slump", and threaten to contribute to a competing project out of retribution, and that the originators and maintainers of an 18+ year project owe you something for showing up and sending in PRs for a couple of years. Transmission has a history much larger than most everyone here, and these demands really communicate only ignorance to that. Private communication is the appropriate channel to work these issues out. If thats not possible, assume it's unwanted. 

Let this behavior speak for itself. Don't invite this behavior into the core permission model of the project. Giving individuals exhibiting low tact and maturity into decision making roles invites problems that will bite the project later. 

Appropriate action frustrated contributors could take: start up a community fork/build that makes improvements and offers these upstream at the cadence you are comfortable with. If upstream is truly abandoned as you would suggest, then you are in a position to continue the project. Individual contribution should be motivated by your desire to improve the program for your own use, not lorde over an existing user base. 

However you approach it though, your actions shouldn't appear in line with the xz social engineering attack. Bombarding the thread with low identity, low reputation anonymous accounts, all parroting the same hostile takeover strategies (sense of urgency, appeal to "community", uniform calls for privilege escalation from low reputation, anonymous accounts, use of accounts that could easily be sock-puppets) just wreaks of malicious entryism and looks like the priority is on privilege escalation on the distribution and user base of the target project. Normalizing this strategy to get what you want makes every important project less secure.

**Comment by nevack** [comment]
@bcomnes I don't like the direction this discussion is taking.

GitHub is a mature platform with numerous security measures in place. There are many rules available to secure a codebase from malicious actions. The XZ backdoor incident was different; it was a hard lesson for anyone maintaining critical software to never give up ALL RIGHTS to a potentially malicious party.

There are some measures available: 
1. Nobody, not even the core team, can push directly to release branches; only PRs are allowed. This rule is already in place.
2. New team members can merge PRs, but not the ones they created themselves. This rule should be established, and new team members must be vetted by the core team.
3. At least two approvals—one from the core team and one from the new team—are required for a PR to be merged.
4. Specify code owners for fine grained control over code to be reviewed by members with specific expertise.
5. The core team can merge any PRs, disregarding other rules (even without approvals, even if the person merging is the author of the PR, etc.).
And so on.

Every respectable project has bus factor in mind and security measures in place.

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-teams-and-people-with-access-to-your-repository

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets

<details>
<summary>/offtopic</summary>

Decided to ask advice from ChatGPT on the matter.

Prompt:
```
Write a detailed plan how to configure a github repository to secure it from evil actions and apply separation of concerns.
```

<details>
<summary>Reply:</summary>

Securing a GitHub repository and applying separation of concerns involves setting up robust access controls, enforcing best practices for code contributions, and using automation to monitor and enforce security measures. Here's a detailed plan:

### 1. **Establish a Governance Model**
   - **Define Roles and Responsibilities**: 
     - **Owners**: Have full administrative rights, including access control, repository settings, and critical actions (e.g., deletion, archiving).
     - **Core Team**: Responsible for approving and merging significant changes. They can manage critical aspects of the repository but with limitations on self-approvals.
     - **Contributors**: Can create branches and submit pull requests (PRs) but cannot push directly to protected branches or merge their own PRs.
     - **Reviewers**: Typically senior members of the team or designated reviewers who provide feedback on PRs.
     - **New Team Members**: New or temporary team members with limited access, who cannot merge or approve their own PRs.

### 2. **Repository Configuration**
   - **Branch Protection Rules**:
     - **Protect the Main/Release Branches**: 
       - Enable branch protection on critical branches (e.g., `main`, `release`) to prevent direct pushes. Require all changes to go through PRs.
       - Enable required status checks before merging. Configure CI/CD pipelines to automatically run tests, security scans, and code quality checks.
       - Require at least two approvals for merging: one from the core team and one from a designated reviewer or another team member.
       - Disable force-pushes to the protected branches to prevent accidental or malicious overwriting of history.
       - Restrict who can dismiss pull request reviews to prevent circumventing the review process.
   - **Enforce Signed Commits**:
     - Require signed commits and tags to ensure that the commits are from verified sources. This helps prevent unauthorized code changes.

### 3. **Access Controls**
   - **Limit Repository Access**:
     - Only give write access to trusted members who need it for their role. Use GitHub Teams to manage groups of users and their permissions.
     - Implement fine-grained personal access tokens (PATs) with minimal permissions required for the user’s tasks. Regularly review and revoke tokens as needed.
     - Enable two-factor authentication (2FA) for all team members to prevent unauthorized access.
   - **Integrate with Identity Providers**:
     - For organizations, consider integrating GitHub with a Single Sign-On (SSO) solution. This allows centralized management of user access and adherence to organizational security policies.

### 4. **Pull Request Workflow**
   - **Set Up a Pull Request Template**:
     - Create a pull request template to guide contributors in providing the necessary context, testing information, and any related documentation or issues.
   - **Require Multiple Reviews**:
     - Configure the repository to require at least two approvals before a PR can be merged, as mentioned earlier. Ensure that at least one approval comes from the core team.
     - New team members should not be allowed to approve or merge their own PRs.
   - **Automate Code Reviews**:
     - Use GitHub Actions or third-party services to automatically run code analysis, linting, and security checks on all PRs. This adds an additional layer of security by catching potential issues before they are merged.
   - **Enforce Review Dismissal Restrictions**:
     - Restrict who can dismiss reviews to prevent bypassing the review process. Only senior team members or core team members should have this permission.

### 5. **Audit and Monitoring**
   - **Enable Repository Audit Logging**:
     - Turn on GitHub's audit logging (for organizations) to monitor all significant activities in the repository, including changes to branch protection rules, pull request actions, and permission modifications.
   - **Set Up Alerts for Security Incidents**:
     - Use GitHub’s built-in security alerts to notify the team of vulnerabilities in dependencies and code. Configure notifications for critical actions, such as changes to branch protection rules or permission modifications.
   - **Regular Security Audits**:
     - Schedule regular security audits of the repository's settings, permissions, and workflow configurations. This can be done quarterly or semi-annually to ensure that all configurations remain aligned with the security policy.

### 6. **Continuous Integration/Continuous Deployment (CI/CD)**
   - **Integrate CI/CD Pipelines**:
     - Ensure that all code changes are automatically tested and validated before merging. Configure pipelines to fail builds for any issues detected during the testing, linting, or security scanning processes.
   - **Deploy Safeguards**:
     - For repositories tied to deployment processes, restrict deployment triggers to specific branches and protect those branches as mentioned earlier.
     - Consider using feature flags to manage deployments, allowing you to control which features are active in production without altering the codebase directly.

### 7. **Backup and Recovery**
   - **Automate Backups**:
     - Regularly back up the repository, including critical branches and tags, to an offsite location or a separate GitHub repository with restricted access.
   - **Develop a Recovery Plan**:
     - Create and document a recovery plan that outlines steps to take in the event of a security breach or accidental deletion. This plan should include procedures for restoring the repository from backups and revoking compromised access tokens.

### 8. **Training and Awareness**
   - **Conduct Security Training**:
     - Provide regular security training for all team members to ensure they understand best practices for repository management and code contribution.
   - **Promote Security Awareness**:
     - Encourage a culture of security awareness by regularly communicating security updates, reminders about best practices, and lessons learned from security incidents.

### 9. **Periodic Reviews and Updates**
   - **Review Access Permissions Regularly**:
     - Conduct periodic reviews of all user access to ensure that only necessary permissions are granted and that users who no longer need access are removed.
   - **Update Branch Protection Rules as Needed**:
     - As the team or project evolves, revisit and adjust branch protection rules to maintain security without hindering productivity.

### 10. **Incident Response**
   - **Set Up an Incident Response Plan**:
     - Have a clear plan for responding to security incidents, including steps for identifying, containing, and mitigating the impact. This plan should also detail communication protocols and post-incident reviews to learn from the event.

By following this plan, you can effectively secure a GitHub repository against malicious actions and maintain a clear separation of concerns among your team members.

</details>

</details>

**Comment by Tharn** [comment]
The funny thing is that if you peruse the Issue tracker for Transmission at all, you run into most of the names in this thread because they have opened Issues, they've participated in discussions. And as it happens, many of these Issues have languished. Just what are you driving at, @bcomnes?

There has been more movement in the last year, which is *great*, and a lot of it is thanks to the people who have been suggested should be given more responsibility. Now they're voicing their concerns that they're running into walls and things could be going better. That should be taken seriously.

**Comment by bcomnes** [comment]
> Just what are you driving at, @bcomnes?

Simple, when dealing with the issue of permission escalation, be aware to communicate and establish trust in a way the Jian Tan construct type could not pull off their social engineering strategy.

> @bcomnes I don't like the direction this discussion is taking.

Yes agreed. Lets leave it to mods.

**Comment by Rukario** [comment]
@bcomnes I don't know if you do but you certainly sound like you have some kind of personal issues with @tearfur. You include every characteristics of @tearful then proceed with implication of the downfall of Transmission.

So unless one of the core team is meant to mark this off as non-issue, it remains an issue. Do so and that core member will show its true color. There are plenty of options worth discussing to do with pacing issues with this project. I prefer that is going to be @ckerr's making a call. For all the fearmongering please be assured I don't think the core team is that dumb. You have nothing to discuss here but for temporary measures if necessary I think it's you that should be muted.

**Comment by bcomnes** [comment]
Absolutely nothing personal. Contributions even look fine. 

My issue is with the absurd and hostile assertion that 3 months between point releases indicates project abandonment.

**Comment by tearfur** [comment]
Usually I refrain from making pessimistic statements like "this project is in a slump". What became the tipping point and elicited a response from me to this thread is the potential implications from the fact that the `4.0.6` tarball FTBFS, which was reported on the day of release, but there has been no word about it at all from the maintainers (see #6873). This was starkly different from previous examples of problems with the release assets, where they were quickly addressed (example #6353).

Personally I don't think the situtation is as bad as "the project is abandoned". After all there are still PRs being merged after `4.0.6` release. Still, I can't help but think "Has the maintainers shifted this much attention away from the project that they couldn't even afford at least acknowledging it?".

Of course that doesn't mean that I can rightfully blurt out these things, but still I'd like to explain myself.

* * *

Moving to the topic of my request for merge access. I've directly mentioned it to the maintainers twice only. Once to make the request, which was met with a lukewarm response, stating that they are not comfortable with it, but "not all hope is lost". Thus there's the second time where I asked if there are any news about it.

The rest of the buzz is from the numerous people who vouch for me, which I'm thankful for, but I'd like to assert that I did not organise this "gaggle" of people, nor am I part of any organised effort to protest for more people with elevated access.

Anyway, to avoid setting more precedents for patterns that make the XZ backdoor incident possible, I don't plan to mention it again any time soon.

**Comment by Pentaphon** [comment]
> My issue is with the absurd and hostile assertion that 3 months between point releases indicates project abandonment coupled to a bottom up privilege escalation request.

Yes, the use of the word "abandoned" was taking it a bit too far. I would completely abandon the use of that word ITT. 

However, people have to start somewhere so it isn't ridiculous for somebody like Tearfur to request write access. What matters is that the admins vet him or anybody else that asks after doing considerable work to help the project. The people in support of Tearfur are simply wishing for the project to move along at a better pace with somebody else at the helm. 

The things I wish to see addressed by the Transmission project are: 

1. Utilization of Github tools like Github topics to make the project more discoverable to people who don't know about the C++ refactoring as discussed in https://github.com/transmission/transmission/issues/6150
2. Stickying important issues to the top of the issues page to get more eyes on them. 
3. Priority merging of very important PRs that work across all clients like https://github.com/transmission/transmission/pull/6105 which adds one of the many features Transmission needs to keep up with clients like Deluge and qBittorrent. 
4. I'd wish the admins would address HUGE PRs like this new compact mode UI https://github.com/transmission/transmission/pull/5319 so that we know what position they have on it instead of keeping us out of the loop. If you don't want it merged, please just tell us?
5. I wish the admins would make an appearance in project management issues made by @Coeur such as https://github.com/transmission/transmission/issues/6952 to show that such issues are being acknowledged.
6. Again, I wish the admins would open a private chat channel with the contributors so they can actually talk to each other and coordinate instead of just leaving Github issues and PRs for them to find.

**Comment by bbsixzz** [comment]
ckerr commented on May 27 https://github.com/transmission/transmission/issues/6813#issuecomment-2132430230
> I agree that something need to happen to make contributions land more smoothly & releases happen more often, but it's not clear to me that there's an obvious easy fix.

Since we're all in agreement on that, how about more communication and less bickering?

**Comment by Coeur** [comment]
> I don't think the torrenting community will benefit from having just one active torrent lib out there. I think libtransmission and libtorrent-rasterbar should both be active.

I agree.
Searching for open-sources alternatives, I found so far, by rough popularity:
1. [libtorrent-rasterbar](https://github.com/arvidn/libtorrent) - used by qBittorrent and Deluge
2. [libtransmission](https://github.com/transmission/transmission) - used by Transmission
3. [libtorrent-rakshasa](https://github.com/rakshasa/libtorrent) - used by rTorrent
4. [libtorrent-swift](https://github.com/XITRIX/LibTorrent-Swift) - used by iTorrent, coming with iOS support

Since my 1.5 years old 7000+ lines of code attempt to partially adopt Swift (#4540) is going into limbo, I could also take a look at that libtorrent-swift effort from XITRIX.

**Comment by reardonia** [comment]
> > Honestly, I can't imagine where @tearfur finds the strength to keep digging into and fixing problems 😞
> 
> @nevack After slogging through the bigger half of this year without much response at all from the maintainers, I now just treat this project as a playground while I can, without too much of a care about making progress (it's not like I have the rights to do so anyway 🤷🏻‍♂️).
> 
> I started contributing to libtransmission because I saw code that's clearly been slowly built up with finely thought out steps. The previous authors strived their best to write out each routine only once, organised them into categories (aka classes in C++, or prefixed functions back in the C days) such that they can be reused to their max. Honestly it's really pretty code in my eyes, so I wanted to ride along the transition phase from C to C++.

I've been around this project for a decade. I'm no longer a professional developer so I haven't maintained the skills to contribute in any substantial ways (ie modern C++ is at best something I decipher as I go), but I do test the sh*t out of stuff and try to code-review what I can. I say this just to emphasize that I'm not an interloper and that I hope the maintainers will consider my vote: @ckerr and @mikedld, I strongly recommend adding @tearfur as a maintainer for libtransmission. He has gone deep into the logic of libtr and is careful and precise in his repairs and introductions of new functionality.

Tr4 (4.0.x) has major bugs and regressions from Tr3, and Tr4.1 is not even close to releasable without ongoing work from tearfur. My assessment is that tearfur is a trustworthy contributor and maintainer.

Another alternative would be for ckerr to communicate a schedule that contributors can rely upon for review and acceptance/rejection of PRs. The current pace of engagement is not viable.  None of this is meant to undermine respect for the awesome work by ckerr and others over 15 years to bring Transmission along.

**Comment by Coeur** [comment]
The admins could also split the project in two repositories:
1. The core as libtransmission
2. The GUI clients as Transmission

That way, they can grant "merge" access to only half the project. And calling it "libtransmission" could attract new projects to adopt it for their own clients.

**Comment by Pentaphon** [comment]
>The admins could also split the project in two repositories:

@Coeur brilliant idea and somebody should have thought of this a long time ago especially considering that the other 2 active open source torrent libraries do the same thing and there is already a client called Fragment that uses libtransmission. Splitting libtransmission into its own repo could even attract more contributors. 

This is absolutely the next course of action to take before anything else can be considered. I really hope this happens in the near future.

**Comment by Krandor1** [comment]
I have to roll back to 7.0.5 due to client being banned.  I would think fixing the libcurl stuff to get the client unbanned would be a priority but I guess not.

**Comment by Pentaphon** [comment]
@mikedld @ckerr thoughts on Coeur's suggestion to split the project into 2 repositories? I think it might be worth considering as a way to get around github's current privilege limitations and a way to attract new contributors.

**Comment by Pentaphon** [comment]
It looks like we we going to need a hotfix 4.0.7 release before 4.1.0 beta can be rolled out due to https://github.com/transmission/transmission/issues/7078 and a few other issues. 

Can we get a 4.0.7 milestone?

**Comment by bbsixzz** [comment]
Radio silence, can one ask what the status quo is without ruffling any feathers?

**Comment by killemov** [comment]
> Radio silence, can one ask what the status quo is without ruffling any feathers?

@tearfur is now part of the team.

Splitting off libtransmission looks like a good idea to me.

Please nix release 4.0.6 and replace with 4.0.5 for now on the main website.

**Comment by Pentaphon** [comment]
> @tearfur is now part of the team.

I didn't see any announcement about that but he is indeed a member of the organization now. Congrats to him. 

> Splitting off libtransmission looks like a good idea to me.

I think the most important issues right now are:

1. Releasing 4.0.7 ASAP to help the users out. 
2. Start merging those 95 PR's in order of priority to chug towards...
3. Begin beta testing 4.1.0 with 2 or 3 beta releases.
4. Plan/start the process of splitting off libtransmission
5. Releasing 4.1.0

**Comment by Coeur** [comment]
> > tearfur is now part of the team.
> 
> I didn't see any announcement about that but he is indeed a member of the organization now. Congrats to him.

He has been for a long time. See my "congratulations" message about it in February 2024: https://github.com/transmission/transmission/pull/6570#issuecomment-1937056349

But despite both nevack and tearfur being members for a long time, they told me they don't have write access. So I don't think the situation has changed yet. To observe when the situation will have really changed, it would need pull request merged by tearfur.

**Comment by Pentaphon** [comment]
> He has been for a long time. See my "congratulations" message about it in February 2024

Hmmm. For some reason I thought you were a member and Tearfur was still a contributor since he's relatively new. 

They should also consider upgrading you to member, tbh. Your enthusiasm and longtime contributions more than merit it.

**Comment by killemov** [comment]
> > > tearfur is now part of the team.
> > 
> > I didn't see any announcement about that but he is indeed a member of the organization now. Congrats to him.
> 
> He has been for a long time. See my "congratulations" message about it in February 2024: [#6570 (comment)](https://github.com/transmission/transmission/pull/6570#issuecomment-1937056349)
> 
> But despite both nevack and tearfur being members for a long time, they told me they don't have write access. So I don't think the situation has changed yet. To observe when the situation will have really changed, it would need pull request merged by tearfur.

My bad for not noticing. But the odd situation of being a member and not having write access threw me for a loop.

> Hmmm. For some reason I thought you were a member and Tearfur was still a contributor since he's relatively new.
> 
> They should also consider upgrading you to member, tbh. Your enthusiasm and longtime contributions more than merit it.

What he said. :sunglasses:

**Comment by jmckillican** [comment]
the fact that 4.0.6 is banned by many trackers for so long now without it being pulled or without a small version update is honestly baffling for a project like this.

**Comment by albino1** [comment]
Just push a hotfix and call it a day.  I have no idea why there's so much hand wringing and debate over tossing in irrelevant bugfixes for what should solely be an emergency hotfix.  Release the hotfix, and if there are other commits you think are worthy of a new release, then start working on that one.  Getting distracted by all of these other problems is a waste of time when Transmission is currently banned on every major tracker.  The hotfix release should have gone out immediately after the issue was first reported, literally anything else can wait.

**Comment by killemov** [comment]
@jmckillican @albino1 You do have a point about priorities and apparently censoring posts, which eats into fixing problems, here is higher on the list. TM is already a somewhat niche client and having the current release version be banned is not helping.

**Comment by odrling** [comment]
> censoring posts

Literally 1984.

Anyway, outsiders (like me, so I’m just as guilty now) tuning into this discussion doesn’t seem really appropriate. Releasing a hotfix won’t close this issue so it is mostly off-topic, it seems more appropriate to me for these comments to be in #7094.

It’s not that this issue has no relation to how the current situation with 4.0.6 is handled by the project, but the previous messages are ignoring what this particular discussion is about. It is more about contributors expressing their frustration with how the project is maintained than about users (who are also frustrated as a result, indeed, but it’s a different and more short-term concern).

(for anyone else interested in this issue, sorry for butting in, I just *had* to side with a dictator for a minute)

**Comment by pmarsden** [comment]
The answer to the original question seems to be "Yes", appearance wise, it looks to be abandoned.
A critical issue has gone unresolved for 4 months now, thats not a sign of an active & maintained project.

**Comment by Pentaphon** [comment]
> The answer to the original question seems to be "Yes", appearance wise, it looks to be abandoned.

I wouldn't go that far considering the last merge happened less than 1 hour ago. "Abandoned" simply isn't true. However, the Transmission project is definitely losing users each time the project is neglected for an extended period of time and the admins need to do more to keep the project going when they are unable to.

**Comment by pmarsden** [comment]
> I wouldn't go that far considering the last merge happened less than 1 hour ago.

Thats why I said "appearance wise" rather than actually.
Few care about the little bitty things being merged, they want the major issue(s) fixed.

Transmission is slowly getting a bad reputation on private sites due to the various 4.0.6 issues.
Once a client gains a bad reputation across sites, its really quite hard to get it back to good again.

> We need a 4.0.7 release and a 4.1.0 beta ASAP.

Not worried about 4.1.0, but a 4.0.7 to fix the major issue(s), Yes.

**Comment by Pentaphon** [comment]
> Transmission is slowly getting a bad reputation on private sites due to the various 4.0.6 issues.

I wouldn't say it is getting a bad reputation since lots of clients on private sites have had to be downgraded, such as qBittorrent, for one issue or another but people are noticing the lack of releases from this project. This should have been solved weeks ago. 

While I appreciate the recent merges, but we need a release ASAP.

**Comment by lilygthb** [comment]
As a non-contributor, reading here, and #7094 , it looks more of a power game, avoiding to create a release, when apparently the community has done the necessary work.
Something along the line of "it's my project, and the more you push me the more I delay the release"

**Comment by killemov** [comment]
@lilygthb No. I will not believe this as a motivation for not making a new release. It would be wise, however, to post a status report and a short term roadmap here or somewhere more public.

**Comment by pmarsden** [comment]
For me, the lack of response, or action, from anyone other than increasingly frustrated users, tells its own story.
These days if anyone asks me, I tell them to stick with 3.0.0, that was (is) highly stable, with none of these issues.
Almost all the private sites I use these days advise people to use qbittorrent (despite the fact it has its own issues).

**Comment by bbsixzz** [comment]
Is there really nothing to be said here? 

Last release was May 30 and tomorrow it's December.

@ckerr 

> This is a bugfix-only release. Everyone's feedback on 4.0.x has been very helpful -- thanks for all the suggestions, bug reports, and pull requests!

What happened to the friendly tone above? I don't get it…

**Comment by bcomnes** [comment]
Looks like we're getting another awesome release soon as we just got an expansive prerelease! Thank you for all the hard work from the core maintainers and direct contributors. 

https://github.com/transmission/transmission/releases/tag/4.1.0-beta.1

It seems like this issue should be closed.

**Comment by zoostorm1** [comment]
> Will 4.1.0 ever be released?

Maybe in 2027, at the rate things move around here.

I'm a member of multiple private and public trackers and I see less and less Transmission users, the majority of peers these days are qbittorrent, uTorrent, Deluge or libtorrent. Transmission used to be fairly common as well, but is slowly vanishing. 

I have largely lost interest in new versions now as well, sticking with 3.0.0 on my older C7 boxes, & 4.0.4 on my R8 boxes. Both are stable for me and do what I need.

**Comment by Pentaphon** [comment]
> Will 4.1.0 ever be released?

Yes, but the leadership needs to make some changes in order to allow the community to play a greater role in pushing the project forward.

**Comment by TorsteinVin** [comment]
Last commit 3 months ago and Ckerr lastly 6 months ago. I think it's slowly being abandoned. :(

**Comment by Pentaphon** [comment]
> Last commit 3 months ago and Ckerr lastly 6 months ago. I think it's slowly being abandoned. :(

It's not technically abandoned until the project is archived and read-only. We have to wait for the leadership to actually do something to move the project forward, and the longer it takes, the less people will be interested, so let's hope they do something soon.

**Comment by Coeur** [comment]
No need to wait more. There are active forks.
https://github.com/users/tearfur/projects/1/views/1

**Comment by TorsteinVin** [comment]
> No need to wait more. There are active forks. [@tearfur's Transmission Fork (view)](https://github.com/users/tearfur/projects/1/views/1)

I couldn't find a download link is it not available yet? Or do i have to build it myself for mac? i'm not that technical.

**Comment by Coeur** [comment]
Ah, you have to build it yourself, @TorsteinVin.
Documentation for building: https://github.com/transmission/transmission/blob/main/docs/Building-Transmission.md
And you follow either of "Building the GTK app with CMake" or "Building the QT app with CMake" (tearfur abandoned the Native app).

**Comment by Pentaphon** [comment]
> No need to wait more. There are active forks. [@tearfur's Transmission Fork (view)](https://github.com/users/tearfur/projects/1/views/1)

Even if that fork eventually becomes a project of its own, it will take a while for it to gain trust in the community, which is why I prefer to try and keep the original project going since the admins have not yet ghosted or archived the project.
